### PR TITLE
VeriCite plugin integration into Assignments, Gradebook, and Speedgrader

### PR DESCRIPTION
**Pull request by baholladay**

### CODE REVIEW

**Comment by baholladay** [comment]
I have added VeriCite to the list of "/plugins". This is how you enable VeriCite for your account. Once enabled, an option will show up in the assignment settings for "inline" or "upload" assignment types. This plugin uses all of the existing turnitin data structures in the tables as requests. The only additional DDL required is a flag on the assignments table in order to differentiate which service is enabled:

alter table assignments ADD vericite_enabled boolean;

Here is a video of the plugin in action: 

https://youtu.be/ttdFkyR1VTM?t=36s  (note that it now uses the plugin configuration instead of the account settings shown in the video)

**Comment by claydiffrient** [comment]
@baholladay Thanks for the PR. I don't see a CLA on file for you.  Could you submit a signed copy of our CLA to opensource@instructure.com? The CLA is found here: https://github.com/instructure/canvas-lms/wiki/ica.pdf

**Comment by baholladay** [comment]
@claydiffrient Thanks... I've sent the CLA. Let me know if you did not receive it

**Comment by claydiffrient** [comment]
@baholladay I just noticed that you submitted the pull request against the `stable` branch.  Can you move it to the `master` branch?

**Comment by baholladay** [comment]
Thanks... Looks like I'll have to do some refactoring to get a new PR in against master

**Comment by baholladay** [comment]
I have created a PR against master:
https://github.com/instructure/canvas-lms/pull/815

File: lib/vericite.rb [hunk 1]
```diff
@@ -0,0 +1,456 @@
...
+          if submission.vericite_data_hash[a.asset_string][:similarity_score].blank?
+            paper_id = a.id
+            paper_title = File.basename(a.display_name, ".*")
+            paper_ext = a.extension
+            paper_type = a.content_type
+            if paper_ext == nil
+              paper_ext = ""
+            end
+            paper_size = 100 # File.size(
+            responses[a.asset_string] = sendRequest(:submit_paper, '2', { :pid => paper_id, :ptl => paper_title, :pext => paper_ext, :ptype => paper_type, :psize => paper_size, :pdata => a.open(), :ptype => '2' }.merge!(opts))
```
Comment (ccutrer) [inline lib/vericite.rb]: ptype is duplicated, which means one or the other is getting ignored

File: lib/vericite.rb [hunk 2]
```diff
@@ -0,0 +1,456 @@
...
+        # do not resubmit if the score already exists
+        if submission.vericite_data_hash[submission.asset_string][:similarity_score].blank?      
+          paper_id = Digest::SHA1.hexdigest submission.plaintext_body
+          paper_ext = "html"
+          paper_title = "InlineSubmission"
+          plain_text = "<html>#{submission.plaintext_body}</html>"
+          paper_type = "text/html"
+          paper_size = plain_text.bytesize
+          
+          responses[submission.asset_string] = sendRequest(:submit_paper, '2', {:pid => paper_id, :ptl => paper_title, :pext => paper_ext, :ptype => paper_type, :psize => paper_size, :pdata => plain_text, :ptype => "1" }.merge!(opts))
```
Comment (ccutrer) [inline lib/vericite.rb]: ditto

Comment (baholladay) [inline lib/vericite.rb]: @ccutrer  This looks like its a left over from code that is no longer used. You can see that the ":ptype" argument is never used is vericite.rb

Do I need to create a new PR? This is what the diff looks like:

```
diff --git a/lib/vericite.rb b/lib/vericite.rb
index fd421ba..2a7a7f8 100644
--- a/lib/vericite.rb
+++ b/lib/vericite.rb
@@ -131,12 +131,11 @@ module VeriCite
             paper_id = a.id
             paper_title = File.basename(a.display_name, ".*")
             paper_ext = a.extension
-            paper_type = a.content_type
             if paper_ext == nil
               paper_ext = ""
             end
             paper_size = 100 # File.size(
-            responses[a.asset_string] = sendRequest(:submit_paper, { :pid => paper_id, :ptl => paper_title, :pext => paper_ext, :ptype => paper_type, :psize => paper_size, :pdata => a.open(), :ptype => '2' }.merge!(opts))
+            responses[a.asset_string] = sendRequest(:submit_paper, { :pid => paper_id, :ptl => paper_title, :pext => paper_ext, :psize => paper_size, :pdata => a.open()}.merge!(opts))
           end
         end
       elsif submission.submission_type == 'online_text_entry' && (asset_string.nil? || submission.asset_string == asset_string)
@@ -144,10 +143,9 @@ module VeriCite
         paper_ext = "html"
         paper_title = "InlineSubmission"
         plain_text = "<html>#{submission.plaintext_body}</html>"
-        paper_type = "text/html"
         paper_size = plain_text.bytesize
 
-        responses[submission.asset_string] = sendRequest(:submit_paper, {:pid => paper_id, :ptl => paper_title, :pext => paper_ext, :ptype => paper_type, :psize => paper_size, :pdata => plain_text, :ptype => "1" }.merge!(opts))
+        responses[submission.asset_string] = sendRequest(:submit_paper, {:pid => paper_id, :ptl => paper_title, :pext => paper_ext, :psize => paper_size, :pdata => plain_text}.merge!(opts))
       else
         raise "Unsupported submission type for VeriCite integration: #{submission.submission_type}"
       end
```

**Pull request closed without merging**